Update to Bevy 0.17 and rapier 0.29 along with dependencies nalgebra, glam

lukasfri opened the pull request:
Hey!

First time contributor. I don't quite know what I'm doing (no this is not in any way a vibecoded PR) but I was starting on a Bevy 0.17 project and needed bevy rapier for 0.17. I thought "how hard could it be" so I worked my way through updating all dependencies and making the necessary changes to get it to compile (including changing depreciations) and now everything seems to work fine (all tests pass). I fully understand if this migration needs changes, but it worked for me and in the meantime it works as a stopgap for anyone who wants a commit they can pin to.

Best wishes

Guelakais reviewed (changes requested):
I see two options here if you want to do it right: split your commits further or squash everything together. On the one hand, your commits are too large. You are combining commits that only deal with documentation (the markdown files) with those that deal with the actual features (the Cargo.toml and *.rs files). This makes your commits very confusing. In addition, it is bad style to tinker with the same file multiple times and create a separate commit each time. Commits and pull requests should always be as small as possible to remain manageable. Here, you have changed the same file (changelog.md) with multiple commits. That is not necessary. Put all commits that deal with this file together. That makes the whole thing much cleaner.

Even though my response is very detailed, I think the pull request is good overall. Keep up the good work!

lukasfri commented:
Hey! Yeah, the PR needs some work before going to the tree, I mostly just wanted to check if there was any interest in merging if I fixed it up, and I needed a quick fix for myself. It seemed rude to just barge in and make a large amount of changes which is why I wanted to gauge interest before sorting it, but if it's welcome I'll clean up the commits.

mnmaita reviewed (commented):
Tested this branch in my project and so far so good! I'm leaving a few comments for you to review. Thanks a lot for creating this PR.

Guelakais commented on bevy_rapier2d/examples/debugdump2.rs:
```diff
@@ -2,6 +2,7 @@
 //! run with:
 //! `cargo run --example debugdump2 > dump.dot && dot -Tsvg dump.dot > dump.svg`
 
+use bevy::app::PostUpdate;
```
Yep. It's part of the prelude. I've tested it

lukasfri commented:
Implemented changes as requested, squashed the commits to be more understandable. First one is still big, but I have no real way of splitting it that I know of bar manually going from the beginning, sorry.

Remaining question is if I should rename `MassModifiedEvent`, `CollisionEvent` and `ContactForceEvent` to end in Message since they are no longer Events. I don't know the depreciation policy in this crate, but creating a type alias with a depreciation notice shouldn't be a problem as far as I'm aware unless there are wishes not to change.

notdanilo commented:
I just tested `bevy_rapier2d = { git = "https://github.com/lukasfri/bevy_rapier", rev = "a5c82f08c86f9fd44430081e7d6ea1a5bb296e8b" }`

and it's working well for Windows, Web and Android 👍🏼

Deniskore commented:
A couple of days ago, I started my project using Bevy and got stuck on the fact that rapier does not support the latest version of Bevy.

What needs to be done to merge it and release a new version? I can allocate some time to help if needed.

I think it makes sense to update Rapier to 0.30.1

lukasfri commented:
There appears to exist a few issues that need to be fixed before merging:

- [ ] @stillonearth regression
- [ ] wasm issue in CI needs fixing.
- [ ] @tylersaunders issue with `ColliderDisabled` needs to be investigated.

I don't have the bandwidth right now to continue doing taking care of those issues myself, but I'll obviously merge any fixes into my branch. My interest in upgrading were limited to a side project I was doing to get it working for myself temporarily. However I've got no problem pressing CI buttons etc.

If you @Deniskore want to tackle the above you're more than welcome.

mnmaita commented:
> @lukasfri this possibly fixes the wasm issue
> 
> `.cargo/config.toml`
> 
> ```toml
> [target.wasm32-unknown-unknown]
> rustflags = ["--cfg", "getrandom_backend=\"wasm_js\""]
> ```

Plus enabling the js feature for `getrandom` in Cargo.toml (if not done already):

```toml
[target.'cfg(all(target_family = "wasm", any(target_os = "unknown", target_os = "none")))'.dependencies]
getrandom = { version = "0.3", features = ["wasm_js"] }
getrandom_02 = { version = "0.2", features = ["js"], package = "getrandom" }
```

The `getrandom_02` line won't be necessary if 0.2 is not in the dependency tree. You should be able to check that by doing `cargo tree | grep getrandom` or something like that.

sebcrozet commented:
@Deniskore If the PR you are mentioning is https://github.com/dimforge/rapier/pull/894, I will check it out tomorrow.
